controllers/imagecollector/imagecollector_controller.go

```diff
@@ -0,0 +1,324 @@
+			return res, err
+		}
+		if _, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil {
+			return reconcile.Result{}, fmt.Errorf("Could not delete completed imagejob")
+		}
+	case eraserv1alpha1.PhaseFailed:
+		log.Info("failed phase")
+		if _, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil {
+			return reconcile.Result{}, fmt.Errorf("Could not delete failed imagejob")
+		}
+	default:
+		log.Info("should not reach this point for imagejob", relevantJobs[0])
+	}
+
+	return ctrl.Result{RequeueAfter: time.Second * 30}, nil
+}
+
+func isNotFound(err error) bool {
+	if err != nil && client.IgnoreNotFound(err) == nil {
+		return true
+	}
+	return false
+}
+
+func (r *Reconciler) handleJobDeletion(ctx context.Context, job *eraserv1alpha1.ImageJob) (ctrl.Result, error) {
+	if job.Status.DeleteAfter == nil {
+		return ctrl.Result{}, nil
+	}
+
+	until := time.Until(job.Status.DeleteAfter.Time)
+	if until > 0 {
+		log.Info("Delaying imagejob delete", "job", job.Name, "deleteAter", job.Status.DeleteAfter)
+		return ctrl.Result{RequeueAfter: until}, nil
+	}
+
+	log.Info("Deleting imagejob", "job", job.Name)
+	err := r.Delete(ctx, job)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) createImageJob(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	job := &eraserv1alpha1.ImageJob{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "imagejob-",
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(imageCollector, schema.GroupVersionKind{Group: "eraser.sh", Version: "v1alpha1", Kind: "ImageCollector"}),
+			},
+		},
+		Spec: eraserv1alpha1.ImageJobSpec{
+			JobTemplate: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:            "collector",
+							Image:           *collectorImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"cpu":    resource.MustParse("7m"),
+									"memory": resource.MustParse("25Mi"),
+								},
+								Limits: corev1.ResourceList{
+									"cpu":    resource.MustParse("8m"),
+									"memory": resource.MustParse("30Mi"),
+								},
+							},
+						},
+					},
+					ServiceAccountName: "eraser-controller-manager",
+				},
+			},
+		},
+	}
+
+	err := r.Create(ctx, job)
+	if err != nil {
+		log.Info("Could not create collector ImageJob")
+		return reconcile.Result{}, err
+	}
+
+	log.Info("Successfully created collector ImageJob", "job", job.Name)
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) updateSharedCRD(ctx context.Context, req ctrl.Request, imageCollector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	imageCollectorList := &eraserv1alpha1.ImageCollectorList{}
+	if err := r.List(ctx, imageCollectorList); err != nil {
```

Comment:
This includes the shared ImageCollector resource, so we need to skip it in the loop

Reply:
The spec of the shared imagecollector is empty up until this point, do we still need to skip it? Assuming we are reset the spec to empty at the start of each collect/scan process?

Reply:
I think it's worth doing because it clarifies intent, and is also a fail-safe. If, for whatever reason, the reconcile runs twice in the ten-minute repeat period, we don't want to modify the `imagecollector-shared` resource based on the current state of the `imagecollector-shared` resource. We just want to update it to reflect the most recent state.